src/web_interface/templates/dependency_graph.html

```diff
@@ -0,0 +1,351 @@
+                    <h5 class="modal-title" id="contextLabel">Additional information</h5>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                </div>
+                <div class="modal-body">
+                    Node:
+                    <p id='node-link'>No node selected.</p>
+                    <br>
+                    File type:
+                    <p id='file-type'>No node selected.</p>
+                </div>
+            </div>
+        </div>
+    </div>
+    <div class="header mb-4" style="word-wrap: break-word">
+        <h3>
+            Dependency Graph for File {{ uid | replace_uid_with_hid }}
+            <br>
+            <span style="font-size: 15px"><strong>UID:</strong> {{ uid }}</span>
+        </h3>
+    </div>
+
+
+
+
+    <div class="row mb-3">
+        <div class="col-md-8">
+            <div id="dependencyGraph"></div>
+        </div>
+        <div class="col-md-4">
+            <div class="header mb-4" style="word-wrap: break-word">
+                <h4 style="text-align: center">Legend</h4>
+            </div>
+            <div id="dependencyGraphLegend"></div>
+            <span style="font-size: 15px">
+                A context menu providing additional information about the nodes is available on right click.
+            </span>
+        </div>
+    </div>
+
+<!--  insert js graph magic here  -->
+<script>
+    var nodes, groups, edges, network, legend_network, allNodes;
+    var highlightActive = false;
+
+    // convenience method to stringify a JSON object
+    function toJSON(obj) {
+        return JSON.stringify(obj, null, 4);
+    }
+
+    var get_nodes = eval( {{ nodes | safe }} )
+    var get_edges = eval( {{ edges | safe }} )
+    var get_groups = eval( {{ groups | safe }} )
+    var my_nodes = []
+    var my_edges = []
+    groups = []
+
+    var graph_container = document.getElementById("dependencyGraph");
+    var legend_container = document.getElementById("dependencyGraphLegend");
+
+    for (i in get_nodes) {
+        my_nodes[i] = {id: get_nodes[i].id, label: get_nodes[i].label, group: get_nodes[i].group, full_file_type: get_nodes[i].full_file_type}
+    }
+    for (i in get_edges) {
+        my_edges[i] = {id: get_edges[i].id, from: get_edges[i].source, to: get_edges[i].target}
+    }
+    for (i in get_groups) {
+        groups.push(get_groups[i])
+    }
+
+    // create an array with nodes
+    nodes = new vis.DataSet();
+    nodes.add(my_nodes);
+
+    // create a legend
+    // and change group colors
+    var legend_nodes = new vis.DataSet();
+    var legend_nodes_array = [];
+    var x = 0;
+    var y = 0;
+    var step = 60;
+
+    var group_options = {};
+    var color_list = []
+    if (typeof {{ color_list | safe }} !== 'undefined' && {{ color_list | safe }}.length > 0) {
+        color_list = {{ color_list | safe }}
```

Comment:
It would be a good idea to put `{{ color_list | safe }}` in a variable, so that it doesn't get rendered three times